src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,72 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final int OPERATION_TYPE_INDEX = 0;
+    private static final int AMOUNT_INDEX = 1;
+    private static final String SPECIFIED_CHARACTER = ",";
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+    private static final String OPERATION_RESULT_TITLE = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String statisticData = readStatisticData(fromFileName);
+        writeStatisticData(toFileName, statisticData);
+    }
+
+    private String readStatisticData(String fromFileName) {
+        File sourceFile = new File(fromFileName);
+        int supplyAmount = 0;
+        int buyAmount = 0;
+        String line;
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(sourceFile))) {
+            do {
+                line = reader.readLine();
+                if (line != null) {
+                    String[] dataRow = line.split(SPECIFIED_CHARACTER);
+                    if (dataRow[OPERATION_TYPE_INDEX].equals(OperationType.supply.name())) {
+                        supplyAmount += Integer.parseInt(dataRow[AMOUNT_INDEX]);
+                    }
+                    if (dataRow[OPERATION_TYPE_INDEX].equals(OperationType.buy.name())) {
+                        buyAmount += Integer.parseInt(dataRow[AMOUNT_INDEX]);
+                    }
+                }
+            } while (line != null);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file: " + sourceFile, e);
+        }
+        return createReport(supplyAmount, buyAmount);
+    }
```

Comment:
Single Responsibility principle mean that your method only read statistic from file but you return ready report, just divide this logic